tensorflow/lite/micro/kernels/batch_matmul_test.cc

```diff
@@ -76,16 +76,19 @@ micro::KernelRunner* GetKernelRunnerInstance(
 }
 
 void TestBatchMatMulFloat(const TfLiteBatchMatMulParams& params,
-                          int* input_dims_data[kNumInputs],
+                          const int* input_dims_data[kNumInputs],
                           const float* input_data_lhs,
-                          const float* input_data_rhs, int* expected_dims,
+                          const float* input_data_rhs, const int* expected_dims,
                           const float* expected_data, float* output_data,
                           bool need_constant_rhs = false,
                           bool need_init_prepare = true) {
-  TfLiteIntArray* input_dims_lhs = IntArrayFromInts(input_dims_data[0]);
-  TfLiteIntArray* input_dims_rhs = IntArrayFromInts(input_dims_data[1]);
-  TfLiteIntArray* output_dims = IntArrayFromInts(expected_dims);
-  const int output_count = ElementCount(*output_dims);
+  TfLiteIntArray* input_dims_lhs =
+      IntArrayFromInts(const_cast<int*>(input_dims_data[0]));
```

Comment:
I didn't realize that this would have such a ripple effect. Generally, I'd consider `const_cast` to be a code smell. How do we know that the dims array won't be modified when we stick it into a `TfLiteTensor`? But we have precedence here, as there are a ton of `const_cast`s in the codebase.

I think we should do one of two things:
1. Revert these back to non-const pointers and adjust the naming of the dims arrays to be snake_case like other non-const variables.
2. Bury the const_cast within `IntArrayFromInts`. Notably, `FloatArrayFromFloats` takes a `const float*` and then casts it away. We could do similar with `IntArrayFromInts` and make it `TfLiteIntArray* IntArrayFromInts(const int*)`.

I think I'd bias towards the first option, but I'm open to opinions here.

Reply:
Tensor dims can never be modified (they are supposed to live in the flatbuffer), without first calling CreateWritableTensorDimsWithCopy on both the TfLiteTensor and the TfLiteEvalTensor.

Reply:
Ok, so really the `TfLiteIntArray* dims` in `TfLiteTensor` should theoretically be `const TfLiteIntArray*`? 

Do you mind trying the 2nd option and seeing if anything blows up? If nothing, then lets ship that and remove the const_casts here.

Reply:
here's an old bug about const_cast that may or may not help: http://b/188459715

Reply:
Updated with 2nd option.